tests/Doctrine/Tests/Persistence/Mapping/Fixtures/TestClassMetadata.php

```diff
@@ -8,25 +8,26 @@
 use LogicException;
 use ReflectionClass;
 
+/**
+ * @template T of object
+ * @implements ClassMetadata<T>
+ */
 final class TestClassMetadata implements ClassMetadata
 {
     /**
      * @var string
-     * @psalm-var class-string
+     * @psalm-var class-string<T>
      */
     private $className;
 
     /**
-     * @psalm-param class-string $className
+     * @psalm-param class-string<T> $className
      */
     public function __construct(string $className)
     {
         $this->className = $className;
     }
 
-    /**
```

Comment:
Why did you remove that?

Reply:
Because this class implements `ClassMetadata` and `ClassMetadata::getName()` already has the proper docblock.

https://github.com/doctrine/persistence/pull/171/files#diff-2d04993be2c5f1b2168b04befaa7921df5ec3816fae3cabd7dd084e2686ede43R20